src/components/ATATFileUpload.vue

```diff
@@ -322,13 +329,21 @@ export default class ATATFileUpload extends Vue {
       if(isRestrictedName){
         this.logInvalidFiles(vFile, doesFileExist);
       }
-      
+
       return isValidFormat && !doesFileExist && isFileSizeValid && !isRestrictedName;
     });
-
+    if(_validFiles.length > this.maxNumberOfFiles){
+      this.moreThanMax = true
+      const errorText = this.maxNumberOfFiles === 1?
+        `Too many files selected. You can upload up to ${this.maxNumberOfFiles} file.`
+        :`Too many files selected. You can upload up to ${this.maxNumberOfFiles} files.`
+      this.$refs.atatFileUpload.errorBucket
+        .push(errorText)
+      this.setErrorMessage()
+      return
```

Comment:
this stops the flow of execution and is returning undefined.
Just doublechecking that this is what you want to do.